### PR TITLE
Range queries

### LINKED ISSUE #217: Non-exact searches for tables
- For number and dates
  - [ ] min
  - [ ] max
  - [ ] equals
- For string
  - [ ] starts with
  - [ ] equals
- For string[]
  - [ ] contains
  - [ ] equal at index

### PR DESCRIPTION
**Pull request by maasencioh**
## Filter by text options
image

## Split by sampleCode and filter options for date
image

closes #217

### CODE REVIEW

File: front/src/components/TableQuery/components/DateColumn.tsx [hunk 1]
```diff
@@ -69,26 +125,45 @@ export default function DateColumn({
     index,
     format,
     dispatch,
-    children,
```
Comment (maasencioh) [inline front/src/components/TableQuery/components/DateColumn.tsx:72]: If I keep the children at the deps, it loops infinitely

File: front/src/components/TableQuery/components/DateColumn.tsx [hunk 2]
```diff
@@ -21,7 +37,47 @@ export default function DateColumn({
...
+        newQuery[`${path}.to`] = limitDate(value, endOfDay);
+        break;
+      }
+    }
+
+    submitQuery({ ...query, ...newQuery, page: '1' });
+  };
+  useEffect(() => {
+    handleChange(value, operator);
+  }, [operator]);
```
Comment (maasencioh) [inline front/src/components/TableQuery/components/DateColumn.tsx]: I use this when I want to update global state after the mutation fo the local state, do you know other way for doing that?

**Review by maasencioh — COMMENTED** [review]
@targos Could you please help me with these points?

File: front/src/components/TableQuery/components/DateColumn.tsx (hunk 2 above)
Comment (targos) [reply front/src/components/TableQuery/components/DateColumn.tsx]: why do you need a state duplication?

File: front/src/components/TableQuery/components/DateColumn.tsx (hunk 1 above)
Comment (targos) [reply front/src/components/TableQuery/components/DateColumn.tsx:72]: There is something very wrong in this component. I'm pretty sure we should never put children in a state.

Comment (maasencioh) [reply front/src/components/TableQuery/components/DateColumn.tsx:72]: this child is a render function, that's been used to format the column content

**Pull request merged**